src/controllers/midi/midicontroller.cpp

```diff
@@ -275,23 +305,53 @@ void MidiController::processInputMapping(const MidiInputMapping& mapping,
             return;
         }
 
-        QJSValue function = pEngine->wrapFunctionCode(mapping.control.item, 5);
-        const auto args = QJSValueList{
-                channel,
-                control,
-                value,
-                status,
-                mapping.control.group,
-        };
-        if (!pEngine->executeFunction(&function, args)) {
-            qCWarning(m_logBase) << "MidiController: Invalid script function"
-                                 << mapping.control.item;
-        }
+        return std::visit(
+                MidiUtils::overloaded{
+                        [this, pEngine, channel, control, value, status](
+                                const ConfigKey& configKey) {
+                            QJSValue function =
+                                    pEngine->wrapFunctionCode(
```

Comment:
I think we should dispose the MidiOption::Script flag and use only the variant type instead. This function wrapper creation can be moved to LegacyMidiControllerMappingFileHandler::load().

Reply:
Maybe I don't have yet a good overview about all, so lets have a look into is a bit more and correct me If I am wrong. 

Here: https://github.com/mixxxdj/mixxx/blob/4e860da4821bfd7c4cd118c42d3d6ee0aac1ec31/src/controllers/midi/legacymidicontrollermappingfilehandler.cpp#L95
The stored xml file is read and the flag `MidiOption::Script` is set, indicating that the `item` element dos not contain a control object but a script function name. 

This PR now adds the feature to add fast direct pointer into the data structure, which should be blazing fast compared to the old method, that calls `wrapFunctionCode()` for every single incoming telegram. My idea was to call `wrapFunctionCode()` only once and than reuse the `QJSValue function` like you did here. 

Therefore wee need a `Controller::getScriptEngine()` access, which is not available in `load()`
Maybe we populate that in `MidiController::setMapping()` or just lazy during `processInputMapping()` itself. 

Does that make sense?